### PR TITLE
Begins Profile components

### PR DESCRIPTION
**Pull request by bartlett705**
This relatively small pull already raises some issues; I tried to break them down a bit to who they concern:

## Front End Team
- Anyone have strong feelings on how we should handle styling? Right now I'm only adding classNames, namespaced by the component that renders them (i.e. 'view-profile-follower' is the classname of a follower span in the 'ViewProfile' component)...that way we can keep our styles somewhat modular with vanilla SASS; I'm definitely down to try one of the solutions for compartmentalizing styles in the actual react component that needs them, but I don't have an opinion on which we should use.

## Back End Team
- We probably need to talk as a whole team about routing principles, but it seems like GET users/:user_id from the browser should probably jump into our app at the view profile state, rather than return a json? Maybe we should put all our api calls in a sub-route (ie `rallypointhost/api/users/:user_id`)?
- The 'user object' in `tests/server/fixtures/userFixtures` is out of sync with the object in `tests/client/fixtures/userObject`...this is because I added some properties that I was relatively sure we would want in the user Objects, and figured that'd be a good way to communicate the front end's needs (i.e. by us writing what we think user objects sent by the server 'should' look like) to the back end team.
- Property names for the mentioned user Objects as returned from the server are not camelCase.  I realize in the db they will be snake_case, but once we get them into JS they should be camelCase, no?  The question is - do we do this property renaming serverside or clientside?  Unless anyone sees why we shouldn't, I'll start writing the client to switch snake_case to camelCase as the server's response is being integrated into the user Object help in the app's state.

### CODE REVIEW

File: src/client/components/Profile.jsx
Comment (mlampedx) [inline src/client/components/Profile.jsx:16]: I think the fetchUser and receiveViewUser functions could be parsed out into separate action creators and reducers to enable serialization of actions and clarify the status of the async process -- e.g. FETCH_USER_REQUEST vs. FETCH_USER_SUCCESS vs. FETCH_USER_FAILURE.

File: src/client/components/Profile.jsx
Comment (mlampedx) [inline src/client/components/Profile.jsx:54]: Should you be referencing user._id and viewUser as props rather than state?

File: src/client/containers/ProfileHolder.jsx
Comment (mlampedx) [inline src/client/containers/ProfileHolder.jsx:11]: Is mapDispatchToProps a placeholder to wrap future action creators? Otherwise, can be removed.

**Review by mlampedx — COMMENTED** [review]
Great work! Left a few questions as comments. Happy to merge after you take a look.

**Comment by mlampedx** [comment]
Regarding camelization of snake_case user objects, I think we should consider using a camelization function as well as a normalizr to handle nested json with ease: https://github.com/paularmstrong/normalizr

File: src/client/components/Profile.jsx
Comment (bartlett705) [reply src/client/components/Profile.jsx:16]: Yep.  That is the plan...those are kind of placeholders since we don't have a store set up / those are the functions that will actually be bound ActionCreators / call bound ActionCreators.

File: src/client/containers/ProfileHolder.jsx
Comment (bartlett705) [reply src/client/containers/ProfileHolder.jsx:11]: Yep!  Just wanted to have the skeleton of a container component there so I could start adding the redux bits.

**Comment by bartlett705** [comment]
RE: casing - yeah, I dig that - you're thinking call both of those functions in the client, on the response data from the server, before adding it to the store?

Maybe this is outside the realm of this discussion - but I've been looking into **isomorphic apps** lately - anybody else into pursuing that here?  My understanding of how it might be applicable here:

-  Server does the initial render for speed/SEO via ReactDOMServer...
- Then attaches the context to that initial document (I see people doing window.INITIAL_STATE), 
- Then serves that with the client bundle
- Which hydrates that context into the redux store on initialization.
- From that point on, the client bundle handles all the rendering, and communicates with the server via JSON.

File: src/client/components/Profile.jsx
Comment (bartlett705) [reply src/client/components/Profile.jsx:54]: Oh yeah, whoops - I was thinking that this would be the only component that cared about / had access to that part of state, but obviously I forgot it's mapStateToProps, not mapStateToState :laughing: 

That kind of makes the update-state function irrelevant too; I'll fix all of this when I actually add action creators.

Comment (bartlett705) [reply src/client/components/Profile.jsx:54]: Oh yeah, whoops - I was thinking that this would be the only component that cared about / had access to that part of state, but obviously I forgot it's mapStateToProps, not mapStateToState :laughing: 

That kind of makes the update-state function irrelevant too; I'll fix all of this when I actually add action creators.

File: src/client/components/Profile.jsx
Comment (mlampedx) [reply src/client/components/Profile.jsx:16]: Gotcha, just wanted to clarify.

File: src/client/components/Profile.jsx
Comment (mlampedx) [reply src/client/components/Profile.jsx:54]: Sounds good!

**Review by mlampedx — APPROVED** [review]

**Comment by mlampedx** [comment]
RE: camelization and normalizr -- we will call the camelization function and use normalizr to ensure that nested JSON is flattened based on pre-written schemas.

RE: isomorphic -- we are all into making this app isomorphic. Could you like to any sample apps that implement isomorphism with the method that you are indicating? Thanks.

**Comment by bartlett705** [comment]
RE: normalizing - right on, makes sense; I just wanted to clarify where that happens: normalize the JSON the server returns somewhere in the RECEIVE_USER reducer before merging with the store, yeh?

Best was probably redux docs, as usual...they actually run redux on the server; not sure we'd need to do that?
- http://redux.js.org/docs/recipes/ServerRendering.html
- https://www.smashingmagazine.com/2016/03/server-side-rendering-react-node-express/
This one is a bit hard to read 'cause they do all kinds of wacky stuff we wouldn't want to do, but it outlines some of the basics if you can look past that part.

**Pull request merged**
